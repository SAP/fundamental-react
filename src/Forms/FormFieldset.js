import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/fieldset.css';

const classnames = classnamesBind.bind(styles);
/** A **FormFieldset** is used to give semantic meaning to a group of elements inside a form (e.g. Billing or Shipping Address).
 * Grouping fields together into a fieldset also provides styling and accessibility benefits. */
const FormFieldset = React.forwardRef(({ children, className, cssNamespace, ...props }, ref) => {

    const formFieldsetClasses = classnames(
        `${cssNamespace}-fieldset`,
        className
    );

    return (
        <fieldset
            {...props}
            className={formFieldsetClasses}
            ref={ref}>
            {children}
        </fieldset>
    );
});

FormFieldset.displayName = 'FormFieldset';

FormFieldset.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string
};

export default withStyles(FormFieldset);
