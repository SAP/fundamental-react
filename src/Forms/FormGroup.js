import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/form-group.css';

const classnames = classnamesBind.bind(styles);

/** A **FormGroup** encapsulates components related to a single form input.
 * At a minimum, it should contain **FormLabel** and input child components. */
const FormGroup = React.forwardRef(({ children, cssNamespace, ...props }, ref) => {

    // Note that we don't want to include the fd-form-group--inline
    // class here because it is only for FormRadioGroup

    return (<div {...props} className={classnames(`${cssNamespace}-form-group`)}
        ref={ref}>{children}</div>);
});

FormGroup.displayName = 'FormGroup';

FormGroup.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node
};

export default withStyles(FormGroup);
