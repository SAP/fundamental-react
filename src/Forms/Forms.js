import PropTypes from 'prop-types';
import React, { Component } from 'react';

// ------------------------------------------------- Form Set -----------------------------------------------
export const FormSet = ({ children, className }) => {
    return (
        <div className={`fd-form__set${className ? ' ' + className : ''}`}>
            {children}
        </div>
    );
};
FormSet.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

// ------------------------------------------------- Form Item -----------------------------------------------
export const FormItem = ({ isCheck, isInline, children, className }) => {
    return (
        <div
            className={`fd-form__item${isInline ? ' fd-form__item--inline' : ''}${
                isCheck ? ' fd-form__item--check' : ''
            }${className ? ' ' + className : ''}`}>
            {children}
        </div>
    );
};
FormItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    isCheck: PropTypes.bool,
    isInline: PropTypes.bool
};

// ------------------------------------------------- Form Label ----------------------------------------------
export const FormLabel = ({ required, children, className, ...props }) => {
    return (
        <label
            className={`fd-form__label${required ? ' is-required' : ''}${
                className ? ' ' + className : ''
            }`}
            {...props}>
            {children}
        </label>
    );
};
FormLabel.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    required: PropTypes.bool
};

// ------------------------------------------------- Form Message ----------------------------------------------
export const FormMessage = ({ type, children, className, ...props }) => {
    return (
        <span
            className={`fd-form__message${type ? '  fd-form__message--' + type : ''}${
                className ? ' ' + className : ''
            }`}
            {...props}>
            {children}
        </span>
    );
};
FormMessage.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    type: PropTypes.oneOf(['', 'error', 'warning', 'help'])
};

// ------------------------------------------------- Form Input ----------------------------------------------
export const FormInput = ({ state, className, ...props }) => {
    return (
        <input
            className={`fd-form__control${state ? ' is-' + state : ''}${
                className ? ' ' + className : ''
            }`}
            {...props} />
    );
};
FormInput.propTypes = {
    className: PropTypes.string,
    state: PropTypes.string
};

// ------------------------------------------------- Form Textarea ----------------------------------------------
export const FormTextarea = ({ children, className, ...props }) => {
    return (
        <textarea
            className={`fd-form__control${className ? ' ' + className : ''}`}
            {...props}>
            {children}
        </textarea>
    );
};
FormTextarea.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

// ------------------------------------------------- Form Fieldset ----------------------------------------------
export const FormFieldset = ({ children, className, ...props }) => {
    return (
        <fieldset
            className={`fd-form__set${className ? ' ' + className : ''}`}
            {...props}>
            {children}
        </fieldset>
    );
};
FormFieldset.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

// ------------------------------------------------- Form Legend ----------------------------------------------
export const FormLegend = ({ children, className, ...props }) => {
    return (
        <legend
            className={`fd-form__legend${className ? ' ' + className : ''}`}
            {...props}>
            {children}
        </legend>
    );
};
FormLegend.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

// ------------------------------------------------- Form Select ----------------------------------------------
export const FormSelect = ({ disabled, children, className, ...props }) => {
    return (
        <select
            className={`fd-form__control${className ? ' ' + className : ''}`}
            {...props}
            disabled={disabled ? true : ''}>
            {children}
        </select>
    );
};
FormSelect.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    disabled: PropTypes.bool
};

// ------------------------------------------------- Form Radio ----------------------------------------------
export class FormRadio extends Component {
    constructor(props) {
        super(props);

        this.state = {
            selectedItem: this.props.defaultChecked
        };
    }

  handleChange = e => {
      this.setState({
          selectedItem: e.currentTarget.value
      });
  };

  render() {
      const { inputs, disabled, isInline } = this.props;
      let result;

      if (isInline) {
          result = inputs.map(inputItem => (
              <div
                  className='fd-form__item fd-form__item--inline fd-form__item--check'
                  key={inputItem.id}>
                  <label className='fd-form__label' htmlFor={inputItem.id}>
                      <input
                          checked={this.state.selectedItem === inputItem.id}
                          className='fd-form__control'
                          disabled={disabled ? true : ''}
                          id={inputItem.id}
                          name={inputItem.name}
                          onChange={this.handleChange}
                          type='radio'
                          value={inputItem.value} />
                      {inputItem.label}
                  </label>
              </div>
          ));
      } else {
          result = inputs.map(inputItem => (
              <div className='fd-form__item fd-form__item--check' key={inputItem.id}>
                  <input
                      checked={this.state.selectedItem === inputItem.id}
                      className='fd-form__control'
                      disabled={disabled ? true : ''}
                      id={inputItem.id}
                      name={inputItem.name}
                      onChange={this.handleChange}
                      type='radio'
                      value={inputItem.value} />
                  <label className='fd-form__label' htmlFor={inputItem.id}>
                      {inputItem.label}
                  </label>
              </div>
          ));
      }
      return <div>{result}</div>;
  }
}
FormRadio.propTypes = {
    defaultChecked: PropTypes.string,
    disabled: PropTypes.bool,
    inputs: PropTypes.array,
    isInline: PropTypes.bool
};
