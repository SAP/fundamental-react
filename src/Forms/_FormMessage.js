import classnamesBind from 'classnames/bind';
import { FORM_MESSAGE_TYPES } from '../utils/constants';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import formMessageStyles from 'fundamental-styles/dist/form-message.css';
import listStyles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind({
    ...listStyles,
    ...formMessageStyles
});

const FormMessage = React.forwardRef(({ children, className, cssNamespace, forPopoverList, type, ...props }, ref) => {

    const formMessageClasses = classnames(
        {
            [`${cssNamespace}-form-message`]: !forPopoverList,
            [`${cssNamespace}-list__message`]: forPopoverList,
            [`${cssNamespace}-form-message--${type}`]: !forPopoverList && !!type,
            [`${cssNamespace}-list__message--${type}`]: forPopoverList && !!type
        },
        className
    );
    return (
        <div
            {...props}
            aria-live='polite'
            className={formMessageClasses}
            ref={ref}
            role='alert'>
            {children}
        </div>
    );
});

FormMessage.displayName = 'FormMessage';

FormMessage.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** if FormMessage is for a popover with list*/
    forPopoverList: PropTypes.bool,
    /** Sets the variation of the component. Primarily used for styling */
    type: PropTypes.oneOf(FORM_MESSAGE_TYPES)
};

export default withStyles(FormMessage);
