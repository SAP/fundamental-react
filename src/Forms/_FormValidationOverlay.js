import classnamesBind from 'classnames/bind';
import { FORM_MESSAGE_TYPES } from '../utils/constants';
import FormMessage from './_FormMessage';
import Popper from '../utils/_Popper';
import PropTypes from 'prop-types';
import withStyles from '../utils/withStyles';
import React, { useState } from 'react';
import styles from 'fundamental-styles/dist/popover.css';

const classnames = classnamesBind.bind(styles);

const FormValidationOverlay = React.forwardRef((
    {
        className,
        cssNamespace,
        control,
        formMessageProps,
        innerRefClassName,
        popperClassName,
        popperProps,
        referenceClassName,
        show,
        validationState,
        wrapperProps,
        ...rest
    }, ref) => {

    let [showValidationMessage, setShowValidationMessage] = useState(show);

    const _handleBlur = () => {
        setShowValidationMessage(false);
    };

    const _handleFocus = () => {
        setShowValidationMessage(true);
    };

    const popoverClasses = classnames(`${cssNamespace}-popover`, className);

    const referenceClasses = classnames(`${cssNamespace}-popover__control`, referenceClassName);

    const bodyContent = (<FormMessage {...formMessageProps} type={validationState?.state}>{validationState?.text}</FormMessage>);

    const referenceComponent = React.cloneElement(control, rest);


    return (
        <div
            {...wrapperProps}
            className={popoverClasses}
            onBlur={_handleBlur}
            onFocus={_handleFocus}
            ref={ref}>
            <Popper
                cssBlock={`${cssNamespace}-popover`}
                innerRefClassName={innerRefClassName}
                noArrow
                popperClassName={popperClassName}
                popperPlacement={'bottom-start'}
                popperProps={popperProps}
                referenceClassName={referenceClasses}
                referenceComponent={referenceComponent}
                show={showValidationMessage && validationState?.text?.length > 0}
                usePortal>
                {bodyContent}
            </Popper>
        </div>
    );
});
FormValidationOverlay.displayName = 'FormValidationOverlay';

FormValidationOverlay.propTypes = {
    /** CSS class(es) to add to the outer wrapping div */
    className: PropTypes.string,
    control: PropTypes.node,
    /** Additional props to be spread to the FormMessage component */
    formMessageProps: PropTypes.object,
    /** CSS class(es) to add to the popper's child `<div>` wrapping the provided children */
    innerRefClassName: PropTypes.string,
    /** CSS class(es) to add to popper's outermost wrapping `<div>` element */
    popperClassName: PropTypes.string,
    /** Additional props to be spread to the overlay element, supported by <a href="https://popper.js.org" target="_blank">popper.js</a> */
    popperProps: PropTypes.object,
    /** CSS class(es) to add to the reference div */
    referenceClassName: PropTypes.string,
    /** Set to **true** to default ValidationOverlay to an open state */
    show: PropTypes.bool,
    /** An object identifying a validation message.  The object will include properties for `state` and `text`; _e.g._, \`{ state: \'warning\', text: \'This is your last warning\' }\` */
    validationState: PropTypes.shape({
        /** State of validation: 'error', 'warning', 'information', 'success' */
        state: PropTypes.oneOf(FORM_MESSAGE_TYPES),
        /** Text of the validation message */
        text: PropTypes.string
    }),
    /** Additional props to be spread to the wrapping `<div>` element */
    wrapperProps: PropTypes.object
};

FormValidationOverlay.defaultProps = {
    popperProps: {}
};

export default withStyles(FormValidationOverlay);
