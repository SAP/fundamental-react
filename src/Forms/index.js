export { default as FormFieldset } from './FormFieldset.js';
export { default as FormInput } from './FormInput.js';
export { default as FormItem } from './FormItem.js';
export { default as FormLabel } from './FormLabel.js';
export { default as FormLegend } from './FormLegend.js';
export { default as FormMessage } from './FormMessage.js';
export { default as FormRadioGroup } from './FormRadioGroup.js';
export { default as FormRadioItem } from './FormRadioItem.js';
export { default as FormSelect } from './FormSelect.js';
export { default as FormSet } from './FormSet.js';
export { default as FormTextarea } from './FormTextarea.js';
