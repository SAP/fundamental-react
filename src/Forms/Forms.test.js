import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { FormFieldset, FormInput, FormItem, FormLabel, FormLegend, FormMessage, FormRadio, FormSelect, FormSet, FormTextarea} from './Forms';

describe('<Forms />', () => {
    const formInput = (
        <FormSet>
            <FormItem isCheck isInline>
                <FormLabel forAttr='input-1'>Default Input</FormLabel>
                <FormInput
                    id='input-1'
                    placeholder='Field placeholder text'
                    type='text' />
            </FormItem>
            <FormItem>
                <FormLabel forAttr='input-1' required>
                    Default Input
                </FormLabel>
                <FormInput
                    id='input-1'
                    placeholder='Field placeholder text'
                    state='help'
                    type='text' />
                <FormTextarea className='blue' id='textarea-1'>
                    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
                </FormTextarea>
                <FormTextarea id='textarea-2'>
                    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
                </FormTextarea>
                <FormMessage type='help'>
                    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
                </FormMessage>
                <FormMessage className='blue'>
                    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
                </FormMessage>
                <FormSelect id='select-1'>
                    <option value='1'>Duis malesuada odio volutpat elementum</option>
                    <option value='2'>Suspendisse ante ligula</option>
                    <option value='3'>Sed bibendum sapien at posuere interdum</option>
                </FormSelect>
                <FormSelect className='blue' disabled
                    id='select-1'>
                    <option value='1'>Duis malesuada odio volutpat elementum</option>
                </FormSelect>
            </FormItem>
        </FormSet>
    );

    const formFieldSet = (
        <FormFieldset>
            <FormLegend className='blue'>Radio buttons</FormLegend>
            <FormRadio
                className='blue'
                defaultChecked='radio-2'
                inputs={[
                    {
                        id: 'radio-1',
                        name: 'radio-1',
                        value: 'radio-1',
                        label: 'Option 1'
                    },
                    {
                        id: 'radio-2',
                        name: 'radio-2',
                        value: 'radio-2',
                        label: 'Option 2'
                    },
                    {
                        id: 'radio-3',
                        name: 'radio-3',
                        value: 'radio-3',
                        label: 'Option 3'
                    }
                ]} />
            <FormLegend legendText='Radio buttons disabled' />
            <FormRadio
                defaultChecked='radio-4'
                disabled
                inputs={[
                    {
                        id: 'radio-4',
                        name: 'radio-4',
                        value: 'radio-4',
                        label: 'Option 1'
                    },
                    {
                        id: 'radio-5',
                        name: 'radio-5',
                        value: 'radio-5',
                        label: 'Option 2'
                    },
                    {
                        id: 'radio-6',
                        name: 'radio-6',
                        value: 'radio-6',
                        label: 'Option 3'
                    }
                ]}
                isInline />
            <FormLegend legendText='Inline Radio buttons' />
            <FormRadio
                defaultChecked='radio-9'
                inputs={[
                    {
                        id: 'radio-7',
                        name: 'radio-7',
                        value: 'radio-7',
                        label: 'Option 1'
                    },
                    {
                        id: 'radio-8',
                        name: 'radio-8',
                        value: 'radio-8',
                        label: 'Option 2'
                    },
                    {
                        id: 'radio-9',
                        name: 'radio-9',
                        value: 'radio-9',
                        label: 'Option 3'
                    }
                ]}
                isInline />
            <FormRadio
                defaultChecked='radio-9'
                disabled
                inputs={[
                    {
                        id: 'radio-7',
                        name: 'radio-7',
                        value: 'radio-7',
                        label: 'Option 1'
                    },
                    {
                        id: 'radio-8',
                        name: 'radio-8',
                        value: 'radio-8',
                        label: 'Option 2'
                    },
                    {
                        id: 'radio-9',
                        name: 'radio-9',
                        value: 'radio-9',
                        label: 'Option 3'
                    }
                ]} />
            <FormLegend legendText='Checkboxes' />
            <FormItem className='blue' isCheck>
                <FormInput
                    className='blue'
                    id='checkbox-1'
                    name='checkbox-name-1'
                    type='checkbox'
                    value='' />
                <FormLabel className='blue' forAttr='checkbox-1'>
                    Option One
                </FormLabel>
            </FormItem>
        </FormFieldset>
    );

    const formSetWithClass = (
        <FormSet className='blue'>
            <FormItem isCheck isInline>
                <FormLabel forAttr='input-1'>Default Input</FormLabel>
                <FormInput
                    id='input-1'
                    placeholder='Field placeholder text'
                    type='text' />
            </FormItem>
        </FormSet>
    );

    const formFieldSetWithClass = (
        <FormFieldset className='blue'>
            <FormLegend className='blue'>Radio buttons</FormLegend>
            <FormRadio
                className='blue'
                defaultChecked='radio-2'
                inputs={[
                    {
                        id: 'radio-1',
                        name: 'radio-1',
                        value: 'radio-1',
                        label: 'Option 1'
                    },
                    {
                        id: 'radio-2',
                        name: 'radio-2',
                        value: 'radio-2',
                        label: 'Option 2'
                    },
                    {
                        id: 'radio-3',
                        name: 'radio-3',
                        value: 'radio-3',
                        label: 'Option 3'
                    }
                ]} />
        </FormFieldset>
    );

    const formRadio = (
        <FormRadio
            defaultChecked='radio-2'
            inputs={[
                {
                    id: 'radio-1',
                    name: 'radio-1',
                    value: 'radio-1',
                    label: 'Option 1'
                },
                {
                    id: 'radio-2',
                    name: 'radio-2',
                    value: 'radio-2',
                    label: 'Option 2'
                },
                {
                    id: 'radio-3',
                    name: 'radio-3',
                    value: 'radio-3',
                    label: 'Option 3'
                }
            ]} />
    );

    test('create form item', () => {
        // create form set with form inputs
        let component = renderer.create(formInput);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // create form ield set with form inputs
        component = renderer.create(formSetWithClass);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // create form ield set with form inputs
        component = renderer.create(formFieldSetWithClass);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // create form ield set with form inputs
        component = renderer.create(formFieldSet);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('change selected radio', () => {
        const wrapper = mount(formRadio);
        expect(wrapper.state('selectedItem')).toBe('radio-2');

        wrapper
            .find('input.fd-form__control[type="radio"]')
            .at(0)
            .simulate('change', { currentTarget: { value: 'radio-1' } });

        expect(wrapper.state('selectedItem')).toBe('radio-1');
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormFieldset component', () => {
            const element = mount(<FormFieldset data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the FormInput component', () => {
            const element = mount(<FormInput data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the FormItem component', () => {
            // TODO: placeholder for this test description once that functionality is built
            const element = mount(<FormItem data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the FormLabel component', () => {
            const element = mount(<FormLabel data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the FormLegend component', () => {
            const element = mount(<FormLegend data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the FormMessage component', () => {
            const element = mount(<FormMessage data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the FormRadio component', () => {
            // TODO: placeholder for this test description once that functionality is built
            const element = mount(<FormRadio data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the FormRadio component\'s input element', () => {
            // TODO: placeholder for this test description once that functionality is built
        });

        xtest('should allow props to be spread to the FormRadio component\'s label element', () => {
            // TODO: placeholder for this test description once that functionality is built
        });

        xtest('should allow props to be spread to the FormRadio component for isInline', () => {
            // TODO: placeholder for this test description once that functionality is built
            const element = mount(<FormRadio data-sample='Sample' isInline />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the FormRadio component\'s input element for IsInline', () => {
            // TODO: placeholder for this test description once that functionality is built
        });

        xtest('should allow props to be spread to the FormRadio component\'s label element for IsInline', () => {
            // TODO: placeholder for this test description once that functionality is built
        });

        test('should allow props to be spread to the FormSelect component', () => {
            const element = mount(<FormSelect data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the FormSet component', () => {
            // TODO: placeholder for this test description once that functionality is built
            const element = mount(<FormSet data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the FormTextarea component', () => {
            const element = mount(<FormTextarea data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
