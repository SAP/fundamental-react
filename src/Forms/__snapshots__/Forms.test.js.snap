// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forms /> create form item 1`] = `
<div
  className="fd-form__set"
>
  <div
    className="fd-form__item fd-form__item--inline fd-form__item--check"
  >
    <label
      className="fd-form__label"
      forAttr="input-1"
    >
      Default Input
      
    </label>
    <input
      className="fd-form__control"
      id="input-1"
      placeholder="Field placeholder text"
      type="text"
    />
  </div>
  <div
    className="fd-form__item"
  >
    <label
      aria-required={true}
      className="fd-form__label"
      forAttr="input-1"
    >
      Default Input
      *
    </label>
    <input
      className="fd-form__control is-help"
      id="input-1"
      placeholder="Field placeholder text"
      type="text"
    />
    <textarea
      className="fd-form__control blue"
      id="textarea-1"
    >
      Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
    </textarea>
    <textarea
      className="fd-form__control"
      id="textarea-2"
    >
      Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
    </textarea>
    <span
      className="fd-form__message fd-form__message--help"
    >
      Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
    </span>
    <span
      className="fd-form__message blue"
    >
      Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
    </span>
    <select
      className="fd-form__control"
      disabled=""
      id="select-1"
    >
      <option
        value="1"
      >
        Duis malesuada odio volutpat elementum
      </option>
      <option
        value="2"
      >
        Suspendisse ante ligula
      </option>
      <option
        value="3"
      >
        Sed bibendum sapien at posuere interdum
      </option>
    </select>
    <select
      className="fd-form__control blue"
      disabled={true}
      id="select-1"
    >
      <option
        value="1"
      >
        Duis malesuada odio volutpat elementum
      </option>
    </select>
  </div>
</div>
`;

exports[`<Forms /> create form item 2`] = `
<div
  className="fd-form__set blue"
>
  <div
    className="fd-form__item fd-form__item--inline fd-form__item--check"
  >
    <label
      className="fd-form__label"
      forAttr="input-1"
    >
      Default Input
      
    </label>
    <input
      className="fd-form__control"
      id="input-1"
      placeholder="Field placeholder text"
      type="text"
    />
  </div>
</div>
`;

exports[`<Forms /> create form item 3`] = `
<fieldset
  className="fd-form__set blue"
>
  <legend
    className="fd-form__legend blue"
  >
    Radio buttons
  </legend>
  <div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={false}
        className="fd-form__control"
        disabled=""
        id="radio-1"
        name="radio-1"
        onChange={[Function]}
        type="radio"
        value="radio-1"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-1"
      >
        Option 1
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={true}
        className="fd-form__control"
        disabled=""
        id="radio-2"
        name="radio-2"
        onChange={[Function]}
        type="radio"
        value="radio-2"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-2"
      >
        Option 2
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={false}
        className="fd-form__control"
        disabled=""
        id="radio-3"
        name="radio-3"
        onChange={[Function]}
        type="radio"
        value="radio-3"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-3"
      >
        Option 3
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`<Forms /> create form item 4`] = `
<fieldset
  className="fd-form__set"
>
  <legend
    className="fd-form__legend blue"
  >
    Radio buttons
  </legend>
  <div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={false}
        className="fd-form__control"
        disabled=""
        id="radio-1"
        name="radio-1"
        onChange={[Function]}
        type="radio"
        value="radio-1"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-1"
      >
        Option 1
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={true}
        className="fd-form__control"
        disabled=""
        id="radio-2"
        name="radio-2"
        onChange={[Function]}
        type="radio"
        value="radio-2"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-2"
      >
        Option 2
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={false}
        className="fd-form__control"
        disabled=""
        id="radio-3"
        name="radio-3"
        onChange={[Function]}
        type="radio"
        value="radio-3"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-3"
      >
        Option 3
      </label>
    </div>
  </div>
  <legend
    className="fd-form__legend"
    legendText="Radio buttons disabled"
  />
  <div>
    <div
      className="fd-form__item fd-form__item--inline fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="radio-4"
      >
        <input
          checked={true}
          className="fd-form__control"
          disabled={true}
          id="radio-4"
          name="radio-4"
          onChange={[Function]}
          type="radio"
          value="radio-4"
        />
        Option 1
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--inline fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="radio-5"
      >
        <input
          checked={false}
          className="fd-form__control"
          disabled={true}
          id="radio-5"
          name="radio-5"
          onChange={[Function]}
          type="radio"
          value="radio-5"
        />
        Option 2
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--inline fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="radio-6"
      >
        <input
          checked={false}
          className="fd-form__control"
          disabled={true}
          id="radio-6"
          name="radio-6"
          onChange={[Function]}
          type="radio"
          value="radio-6"
        />
        Option 3
      </label>
    </div>
  </div>
  <legend
    className="fd-form__legend"
    legendText="Inline Radio buttons"
  />
  <div>
    <div
      className="fd-form__item fd-form__item--inline fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="radio-7"
      >
        <input
          checked={false}
          className="fd-form__control"
          disabled=""
          id="radio-7"
          name="radio-7"
          onChange={[Function]}
          type="radio"
          value="radio-7"
        />
        Option 1
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--inline fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="radio-8"
      >
        <input
          checked={false}
          className="fd-form__control"
          disabled=""
          id="radio-8"
          name="radio-8"
          onChange={[Function]}
          type="radio"
          value="radio-8"
        />
        Option 2
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--inline fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="radio-9"
      >
        <input
          checked={true}
          className="fd-form__control"
          disabled=""
          id="radio-9"
          name="radio-9"
          onChange={[Function]}
          type="radio"
          value="radio-9"
        />
        Option 3
      </label>
    </div>
  </div>
  <div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={false}
        className="fd-form__control"
        disabled={true}
        id="radio-7"
        name="radio-7"
        onChange={[Function]}
        type="radio"
        value="radio-7"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-7"
      >
        Option 1
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={false}
        className="fd-form__control"
        disabled={true}
        id="radio-8"
        name="radio-8"
        onChange={[Function]}
        type="radio"
        value="radio-8"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-8"
      >
        Option 2
      </label>
    </div>
    <div
      className="fd-form__item fd-form__item--check"
    >
      <input
        checked={true}
        className="fd-form__control"
        disabled={true}
        id="radio-9"
        name="radio-9"
        onChange={[Function]}
        type="radio"
        value="radio-9"
      />
      <label
        className="fd-form__label"
        htmlFor="radio-9"
      >
        Option 3
      </label>
    </div>
  </div>
  <legend
    className="fd-form__legend"
    legendText="Checkboxes"
  />
  <div
    className="fd-form__item fd-form__item--check blue"
  >
    <input
      className="fd-form__control blue"
      id="checkbox-1"
      name="checkbox-name-1"
      type="checkbox"
      value=""
    />
    <label
      className="fd-form__label blue"
      forAttr="checkbox-1"
    >
      Option One
      
    </label>
  </div>
</fieldset>
`;
